src/components/application_manager/test/commands/mobile/system_request_test.cc

```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
+  const std::string binary_data_folder;
+  const std::string app_full_file_path;
+  std::vector<uint8_t> binary_data_;
 };
 
 TEST_F(SystemRequestTest, Run_HTTP_FileName_no_binary_data_REJECTED) {
   PreConditions();
-  MessageSharedPtr msg = CreateIVSUMessage();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
 
   ExpectManageMobileCommandWithResultCode(mobile_apis::Result::REJECTED);
 
-  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
-  command->Run();
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_HttpRequestTypeAndExistsFileName_SuccessResponce) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  am::AppFile file;
+  file.is_download_complete = true;
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(mock_policy_handler_, ReceiveMessageFromSDK(kFileName, _));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_RequestType_SuccessRequest) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestClimate))
+      .WillOnce(Return(true));
+
+  am::AppFile file;
+  file.is_download_complete = true;
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::BasicCommunication_SystemRequest)));
+
+  Run(CreateIVSUMessage(kRequestClimate));
+}
+
+TEST_F(SystemRequestTest, Run_NonExistentApp_ApplicationNotRegisteredResponse) {
+  mock_app_.reset();
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+
+  ExpectManageMobileCommandWithResultCode(
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_EmptyParams_DisallowedRequestType_DisallowedResponse) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(false));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::DISALLOWED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileNameSyntax_InvalidData) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] =
+      kFileName + "\n";
+
+  Run(message);
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileName_InvalidData) {
```

Comment:
@BSolonenko I believe this unit test covers the same code as `Run_InvalidFileNameSyntax_InvalidData`. Isn't it?